R/ShinyArchRExports.R

```diff
@@ -0,0 +1,620 @@
+    for(matName in allMatrices){
+      if(matName %in% supportedMatrices){
+        
+          featuresNames <- getFeatures(ArchRProj = ArchRProj, useMatrix = matName)
+          dir.create(file.path(projDir,mainDir, subOutDir, matName), showWarnings = FALSE)
+          saveRDS(featuresNames, file.path(projDir,mainDir, subOutDir, matName, paste0(matName, "_names.rds")))
+          
+          if(!is.null(featuresNames)){
+            
+            mat = Matrix(.getMatrixValues(
+              ArchRProj = ArchRProjShiny, 
+              name = featuresNames,
+              matrixName = matName,
+              log2Norm = FALSE,
+              threads = threads), sparse = TRUE)
+            
+            matrices[[matName]] = mat
+            matList = mat[,rownames(df), drop=FALSE]
+            .logThis(matList, paste0(matName,"-Before-Impute"), logFile = logFile)
+            
+            if(getArchRVerbose()) message("Imputing Matrix")
+            imputeMat <- imputeMatrix(mat = as.matrix(matList), imputeWeights = imputeWeights, logFile = logFile)
+            
+            if(!inherits(imputeMat, "matrix")){
+              imputeMat <- mat(imputeMat, ncol = nrow(df))
+              colnames(imputeMat) <- rownames(df)
+            }
+            imputeMatrices[[matName]] <- imputeMat
+            
+      }else{
+            message(matName, " is NULL.")
+          }
+      }
+    }
+    
+    matrices$allColorBy= .availableArrays(head(getArrowFiles(ArchRProj), 2))
+    saveRDS(matrices, file.path(projDir,mainDir, subOutDir, "matrices.rds"))
+    saveRDS(imputeMatrices, file.path(projDir,mainDir, subOutDir, "imputeMatrices.rds"))
+  }else{
+    
+    message("matrices and imputeMatrices already exist. reading from local files...")
+    
+    matrices <- readRDS(file.path(projDir, mainDir, subOutDir, "matrices.rds"))
+    imputeMatrices <- readRDS(file.path(projDir, mainDir, subOutDir, "imputeMatrices.rds"))
+  }
+  
+  print("Mainembeds started...")
+  
+  # mainEmbeds will create an HDF5 file containing the nativeRaster vectors for data stored in cellColData
+  if (!file.exists(file.path(projDir, mainDir, subOutDir, "mainEmbeds.h5"))) {
+    .mainEmbeds(ArchRProj = ArchRProjShiny,
+                outDirEmbed = file.path(projDir, mainDir, subOutDir),
+                colorBy = "cellColData",
+                cellColEmbeddings = cellColEmbeddings,
+                embeddingDF = df,
+                matrices =  matrices,
+                imputeMatrices = imputeMatrices,
+                logFile = createLogFile("mainEmbeds")
+    )          
+  } else{
+    message("H5 for main embeddings already exists...")
+  }
+  
+  print("MatrixEmbeds started...")
+  
+  # matrixEmbeds will create an HDF5 file containing he nativeRaster vectors for data stored in matrices
+  if(!file.exists(file.path(projDir,mainDir, subOutDir, "plotBlank72.h5"))){
```

Comment:
I was using the naming from your example but I can change it to `"nativeRasters.h5"`. What do you think? They're blank because they don't show the axis and 72 because that was the resolution.